pkg/prometheus/promcfg.go

```diff
@@ -3096,6 +3091,14 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		cfg = append(cfg, yaml.MapItem{Key: "metric_relabel_configs", Value: generateRelabelConfig(labeler.GetRelabelingConfigs(sc.TypeMeta, sc.ObjectMeta, sc.Spec.MetricRelabelConfigs))})
 	}
 
+	if sc.Spec.RelabelConfigs != nil {
+		relabelings = append(relabelings, generateRelabelConfig(labeler.GetRelabelingConfigs(sc.TypeMeta, sc.ObjectMeta, sc.Spec.RelabelConfigs))...)
+	}
+
+	if len(relabelings) > 0 {
```

Comment:
Since `relabelings` is never 0, it is initialized with `initRelabelings()`, would it be fine to not even check the array length?

Reply:
I thought about that and went with the defensive option. But Prometheus wouldn't mind if `relabelings: null` so let's go with the simpler option.